### PR TITLE
[TD][bots] Auto add TD rollout label

### PR DESCRIPTION
**Pull request by clee2000**
Corresponds to https://github.com/pytorch/pytorch/pull/122976

### CODE REVIEW

File: torchci/lib/bot/autoLabelBot.ts [hunk 1]
```diff
@@ -416,6 +438,20 @@ function myBot(app: Probot): void {
         }
       }
 
+      if (
+        isPyTorchPyTorch(owner, repo) &&
+        context.payload.action === "opened"
+      ) {
+        // Add the ciflow/trunk label to PRs opened by authors listed in the TD
```
Comment (huydhn) [inline torchci/lib/bot/autoLabelBot.ts:445]: Minor nit:  I guess you mean add the `ci-td-distributed` label

**Review by huydhn — APPROVED** [review]

File: torchci/lib/bot/autoLabelBot.ts [hunk 2]
```diff
@@ -175,7 +176,28 @@ function getRepoSpecificLabels(
   return repoSpecificAutoLabels[repoKey];
 }
 
+function TDRolloutIssueParser(rawSubsText: string): object {
+  const subsText = rawSubsText.replace("\r", "");
+  const subsRows = subsText.match(/^\*.+/gm);
```
Comment (ZainRizvi) [inline torchci/lib/bot/autoLabelBot.ts:181]: Nit: How about supporting both '*' and '-' style of bullet points? Folks can self edit the issue and may fall back to muscle memory without realizing it

File: torchci/lib/bot/autoLabelBot.ts [hunk 3]
```diff
@@ -175,7 +176,28 @@ function getRepoSpecificLabels(
...
+  const authors: any = new Set();
+  if (subsRows == null) {
+    return authors;
+  }
+  subsRows.forEach((row: string) => {
+    const users = row.match(/@[a-zA-Z0-9-/]+/g);
```
Comment (ZainRizvi) [inline torchci/lib/bot/autoLabelBot.ts:187]: [This](https://github.com/pytorch/pytorch/issues/123120) is the issue it parses, right?  I'm not seeing "@" before a person's name. 

Does the issue need to be fixed?

**Review by ZainRizvi — APPROVED** [review]

File: torchci/lib/bot/autoLabelBot.ts (hunk 3 above)
Comment (clee2000) [reply torchci/lib/bot/autoLabelBot.ts:187]: fixed

File: torchci/lib/bot/autoLabelBot.ts (hunk 2 above)
Comment (clee2000) [reply torchci/lib/bot/autoLabelBot.ts:181]: Github automatically puts another * when you press enter on a line that has a * bullet point

**Pull request merged**